src/bin/batch_solve.rs

```diff
@@ -0,0 +1,209 @@
+    #[arg(long)]
+    ip_range: Option<String>,
+}
+
+#[derive(Debug, clap::Args)]
+#[group(required = true, multiple = false)]
+struct Boards {
+    /// Path to a file containing a list of boards
+    #[clap(long)]
+    boards_file: Option<String>,
+
+    /// Specify the boards on command line
+    #[clap(long, num_args=1..)]
+    boards: Option<Vec<String>>,
+}
+
+fn main() -> Result<(), String> {
+    let args = Args::parse();
+
+    let boards = if let Some(boards) = args.boards.boards {
+        boards
+    } else {
+        let boards_files = args
+            .boards
+            .boards_file
+            .expect("Must specify boards or boards_file");
+        let boards_contents =
+            std::fs::read_to_string(boards_files).expect("Unable to read boards_file");
+        boards_contents
+            .lines()
+            .map(|s| s.to_string())
+            .collect::<Vec<String>>()
+    };
+    let (mut card_config, tree_config) =
+        deserialize_configs_from_file(&args.config).expect("Couldn't deserialize config");
+
+    // Update card_config and tree_config with command-line specified data
+    if let Some(range_string) = args.oop_range {
+        let range_result = range_string.parse::<Range>();
+        if let Ok(range) = range_result {
+            card_config.range[0] = range;
+        } else {
+            println!("Couldn't parse OOP Range \"{}\"", range_string);
+            println!("{}", range_result.unwrap_err());
+        }
+    }
+    if let Some(range_string) = args.ip_range {
+        let range_result = range_string.parse::<Range>();
+        if let Ok(range) = range_result {
+            card_config.range[1] = range;
+        } else {
+            println!("Couldn't parse IP Range \"{}\"", range_string);
+            println!("{}", range_result.unwrap_err());
+        }
+    }
+
+    let max_num_iterations = args.max_iterations;
+    let target_exploitability = tree_config.starting_pot as f32 * args.exploitability;
+    println!("Starting pot: {}", tree_config.starting_pot);
+    println!("Effective stacks: {}", tree_config.effective_stack);
+    println!(
+        "Exploitable for {}% of staring pot ({} chips)",
+        args.exploitability * 100.0,
+        target_exploitability
+    );
+
+    // Create output directory if needed. Check if ".pfs" files exist, and if so abort
+    let dir = PathBuf::from(args.dir);
+    setup_output_directory(&dir)?;
+
+    // Save config to output directory
+    let config_json_path = dir.join("config.json");
+    if config_json_path.exists() {
+        println!(
+            "Config already exists at path {}. Exiting.",
+            config_json_path.display()
+        );
+        exit(1);
+    }
+    let boards_file_out_path = dir.join("boards.txt");
+    if boards_file_out_path.exists() {
+        println!(
+            "Boards file exists at path {}. Exiting.",
+            boards_file_out_path.display()
+        );
+        exit(1);
```

Comment:
Feel like we should overwrite `boards.txt` and `config.json` when `--overwrite` is specified.